controllers/admin/AdminImportController.php

```diff
@@ -1569,80 +2153,88 @@ protected function productImportAccessories($offset, $limit, &$crossStepsVariabl
-                Db::getInstance()->delete('accessory', 'id_product_1 = '.(int)$product_id);
-                Product::changeAccessoriesForProduct($unique_ids, $product_id);
+                $uniqueIds = Product::getExistingIdsFromIdsOrRefs($links);
+                Db::getInstance()->delete('accessory', 'id_product_1 = '.(int)$productId);
+                Product::changeAccessoriesForProduct($uniqueIds, $productId);
             }
-            $line_count++;
+            $lineCount++;
 
-            // Empty value to reduce array weight (that goes through HTTP requests each time) but do not unset array entry!
-            $accessories[$product_id] = 0; // In JSON, 0 is lighter than null or false
+            // Empty value to reduce array weight (that goes through HTTP requests each time).
+            // But do not unset array entry!
+            // In JSON, 0 is lighter than null or false
+            $accessories[$productId] = 0;
 
             // stop when limit reached
-            if ($line_count >= $limit) {
+            if ($lineCount >= $limit) {
                 break;
             }
         }
 
-        if ($line_count < $limit) { // last pass only
+        if ($lineCount < $limit) { // last pass only
             Module::processDeferedFuncCall();
             Module::processDeferedClearCache();
         }
 
         $crossStepsVariables['accessories'] = $accessories;
 
-        return $line_count;
+        return $lineCount;
     }
 
-    protected function productImportOne($info, $default_language_id, $id_lang, $force_ids, $regenerate, $shop_is_feature_active, $shop_ids, $match_ref, &$accessories, $validateOnly = false)
-    {
-        if ($force_ids && isset($info['id']) && (int)$info['id']) {
-            $product = new Product((int)$info['id']);
-        } elseif ($match_ref && array_key_exists('reference', $info)) {
-            $datas = Db::getInstance()->getRow('
-					SELECT p.`id_product`
-					FROM `'._DB_PREFIX_.'product` p
-					'.Shop::addSqlAssociation('product', 'p').'
-					WHERE p.`reference` = "'.pSQL($info['reference']).'"
-				', false);
-            if (isset($datas['id_product']) && $datas['id_product']) {
-                $product = new Product((int)$datas['id_product']);
-            } else {
-                $product = new Product();
+    protected function productImportOne(
+        $productData,
+        $defaultLanguageId,
+        $idLang,
+        $forceIds,
+        $regenerate,
+        $shopIsFeatureActive,
+        $shopIds,
+        $matchRef,
+        &$accessories,
+        $validateOnly = false
+    ) {
+
+        $productId = !empty($productData['id']) ? (int)$productData['id'] : 0;
+
+        if ($matchRef && !empty($productData['reference'])) {
+            $data = Db::getInstance()->getRow(
+                'SELECT p.`id_product`'
+                . ' FROM `' . _DB_PREFIX_ . 'product` p ' . Shop::addSqlAssociation('product', 'p')
```

Comment:
Seen while reviewing #8255, I guess `Shop::addSqlAssociation('product', 'p')` is useless